presto-native-execution/presto_cpp/main/PrestoTask.cpp

```diff
@@ -117,45 +118,126 @@ static void addRuntimeMetric(
+        "{}.{}.{}", op.operatorType, op.planNodeId, std::get<0>(spillMetric));
+    opOut.runtimeStats.emplace(
+        statName,
+        createProtocolRuntimeMetric(
+            statName, std::get<1>(spillMetric), std::get<2>(spillMetric)));
+  }
+}
+
+static void addTaskRuntimeMetrics(
+    std::unordered_map<std::string, RuntimeMetric>& taskRuntimeStats,
+    const exec::OperatorStats& op) {
+  const std::unordered_map<std::string, uint64_t> opMetrics{
+      {"numSplits", op.numSplits},
+      {"inputBatches", op.inputVectors},
+      {"outputBatches", op.outputVectors}};
+
+  for (const auto& [name, value] : opMetrics) {
+    const auto statName =
+        fmt::format("{}.{}.{}", op.operatorType, op.planNodeId, name);
+    taskRuntimeStats.emplace(
+        statName, createVeloxRuntimeMetric(value, RuntimeCounter::Unit::kNone));
+  }
+}
+
+static void addDriverTaskRuntimeMetrics(
+    protocol::TaskStats& prestoTaskStats,
+    const exec::TaskStats& taskStats) {
+  // Task runtime metrics for driver counters.
+  std::unordered_map<std::string, uint64_t> taskMetrics{
+      {"drivers.total", taskStats.numTotalDrivers},
+      {"drivers.running", taskStats.numRunningDrivers},
+      {"drivers.completed", taskStats.numCompletedDrivers},
+      {"drivers.terminated", taskStats.numTerminatedDrivers}};
+  for (const auto it : taskStats.numBlockedDrivers) {
+    taskMetrics.emplace(
+        fmt::format("drivers.{}", exec::blockingReasonToString(it.first)),
+        it.second);
+  }
+
+  for (const auto& [name, value] : taskMetrics) {
+    if (value > 0) {
+      prestoTaskStats.runtimeStats.emplace(
+          name, createProtocolRuntimeMetric(name, value));
+    }
+  }
+}
+
+static void processOperatorStats(
+    protocol::TaskStats& prestoTaskStats,
+    protocol::TaskState state) {
+  bool skipRuntimeStats =
+      SystemConfig::instance()->skipRuntimeStatsInRunningTaskInfo() &&
+      !isFinalState(state);
+
+  if (skipRuntimeStats) {
+    for (auto& pipelineOut : prestoTaskStats.pipelines) {
+      for (auto& opOut : pipelineOut.operatorSummaries) {
+        opOut.runtimeStats.clear();
+      }
+    }
+  } else {
+    const std::vector<std::string> prefixToExclude{"running", "blocked"};
+    for (auto& pipelineOut : prestoTaskStats.pipelines) {
+      for (auto& opOut : pipelineOut.operatorSummaries) {
+        for (const auto& prefix : prefixToExclude) {
+          for (auto it = opOut.runtimeStats.begin();
+               it != opOut.runtimeStats.end();) {
+            if (it->first.find(prefix) != std::string::npos) {
+              it = opOut.runtimeStats.erase(it);
+            } else {
+              ++it;
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+static void processTaskStats(
+    protocol::TaskStats& prestoTaskStats,
+    const velox::exec::TaskStats& veloxTaskStats,
+    const std::unordered_map<std::string, RuntimeMetric>& taskRuntimeStats,
+    protocol::TaskState state) {
```

Comment:
There are a lot of *TaskStats* variables - hard to understand which one holds what.
Can the variable names be changed to reflect this?